discovery/consul/consul.go

```diff
@@ -238,16 +237,20 @@ func (d *Discovery) shouldWatchFromName(name string) bool {
 // *all* services. Details in https://github.com/prometheus/prometheus/pull/3814
 func (d *Discovery) shouldWatchFromTags(tags []string) bool {
 	// If there's no fixed set of watched tags, we watch everything.
-	if d.watchedTag == "" {
+	if len(d.watchedTags) == 0 {
 		return true
 	}
 
+	a := struct{}{}
+	found := make(map[string]struct{})
```

Comment:
This seems a bit inefficient, and won't work if a tag is repeated for some reason. What I'd do is have the inner loop continue the outer if it's found, and return false if it's not.

Reply:
Repeated tags is a weird case. I would assume that looking for the same tag multiple times is a configuration issue, and not something people would actually have on purpose.

But yes, the continue you're suggesting is better :+1: